lib/event_sourceable/event_store.rb

```diff
@@ -73,7 +75,7 @@ def apply_event_and_save
 
       # record the metadata
       if event.respond_to?(:metadata)
-        self.metadata.merge!(event.metadata || {})
+        self.metadata = metadata.merge!(event.metadata || {})
```

Comment:
Should this be `merge`? `merge!` changes in place.

Reply:
I'd `merge` here without the bang